homeassistant/components/ipma/weather.py

```diff
@@ -100,7 +102,12 @@ async def async_setup_entry(
     mode = config_entry.data[CONF_MODE]
 
     api = await async_get_api(hass)
-    location = await async_get_location(hass, api, latitude, longitude)
+    try:
+        location = await async_get_location(hass, api, latitude, longitude)
+    except IPMAException as err:
+        raise ConfigEntryNotReady(
```

Comment:
Raising `ConfigEntryNotReady` here doesn't work. Platforms can raise `PlatformNotReady`.

Did you test this PR?

Reply:
I misread the documentation, I've corrected by moving to __init__ and added tests